pkg/operator/apiserver/controller/apiservice/apiservice_controller.go

```diff
@@ -152,11 +156,18 @@ func (c *APIServiceController) sync(ctx context.Context, syncCtx factory.SyncCon
 	return err
 }
 
-func (c *APIServiceController) syncAPIServices(ctx context.Context, apiServices []*apiregistrationv1.APIService, recorder events.Recorder) error {
+func (c *APIServiceController) syncAPIServices(ctx context.Context, enabledApiServices, disabledApiServices []*apiregistrationv1.APIService, recorder events.Recorder) error {
 	errs := []error{}
 	var availableConditionMessages []string
 
-	for _, apiService := range apiServices {
+	for _, apiService := range disabledApiServices {
+		if err := c.apiregistrationv1Client.APIServices().Delete(ctx, apiService.Name, metav1.DeleteOptions{}); err != nil && !apierrors.IsNotFound(err) {
```

Comment:
I think we should get an APIService from the `cache` first as that would allow us to put less pressure on the server. 
We should issue a live request only when we know the resource must be deleted.
What do you think?

Reply:
Updated